### ISSUE TITLE
Handled a DebuggableError

### ISSUE THREAD

**Issue by ViktorChernykh** [vapor/vapor#2877]
### Describe the bug

When an error is handled as a DebuggableError in the Logger+report file, the log function is called, ignoring source. Because of this, there is no way to log #file, #function, #line. To fix this, you need to call another function from swift-log:
public func log(level: Logger.Level,
                     _message: @autoclosure() -> Logger.Message,
                     metadata: @autoclosure() -> Logger.Metadata? = nil,
                     source: @autoclosure() -> String? = nil,
                     file: String = #fileID, function: String = #function, line: UInt = #line)

* Vapor Framework version: 4.65.1
* Vapor Toolbox version:
* OS version: macOS Monterey

### Additional context
Maybe I don't know how to cook properly?

**Comment by 0xTim** [comment]
I'm confused - what's the problem? We do pass the source through for DebuggableErrors https://github.com/vapor/vapor/blob/main/Sources/Vapor/Logging/Logger%2BReport.swift#L22

**Comment by ViktorChernykh** [comment]
Yes, but it does not reach the log https://github.com/vapor/vapor/blob/main/Sources/Vapor/Logging/Logger%2BReport.swift#L42

**Comment by 0xTim** [comment]
Right I see. Is this actually causing an issue? Because we're logging file, function and line

**Comment by ViktorChernykh** [comment]
The fact is that in the swift-log, when printing, this information is ignored, and only the source is displayed. https://github.com/apple/swift-log/blob/main/Sources/Logging/Logging.swift#L1140

**Comment by bplattenburg** [comment]
If this is still something we're looking to add, I'd like to take a stab at it.

@ViktorChernykh is just updating `Logger+Report` enough to solve your issue? Or does it need to be plumbed through additional callsites e.g. `ErrorMiddleWare.default`?

**Comment by bplattenburg** [comment]
@ViktorChernykh  I started this in a PR, but I realized I need a clearer example of the call path that caused the issue to know where all to plumb the `source` through. Can you share more about your use case and where you ran into this?

I also think the swift-log code you linked out to has changed since the time you linked it, is this the line now? https://github.com/apple/swift-log/blob/f4f84c9f264fc585d2988e1c9fa8c70e80da1662/Sources/Logging/Logging.swift#L228
